specs/agents/otel-distribution.md

```diff
@@ -0,0 +1,77 @@
+
+## Terminology
+
+**Vanilla OpenTelemetry Distribution**: this is the "upstream" OpenTelemetry distribution that is maintained by the OpenTelemetry community.
+Implementation differs per platform, but it usually consists of an API/SDK and can also provide automatic instrumentation.
+
+**Elastic OpenTelemetry Distribution**: this is an OpenTelemetry distribution provided by Elastic that is derived from the 
+_Vanilla OpenTelemetry Distribution_.
+
+## General guidelines
+
+These statements are guiding principles of the Elastic OpenTelemetry distributions, they should be considered more as advice than strict rules.
+
+Elastic OpenTelemetry distribution SHOULD ideally:
+- behave as drop-in replacements of their upstream counterparts
+- provide a simple setup and favor onboarding experience (aka "things should work by default").
+- avoid capturing potentially confusing data (see [system metrics](#system-metrics) example below).
+
+## Configuration
+
+Elastic OpenTelemetry distributions MAY override the default configuration.
+When doing so, user-configuration should remain consistent with vanilla distribution:
+- explicit user configuration SHOULD remain effective
+- overriden default configuration MUST have the ability to be restored to upstream default
+
+Elastic specific configuration items MUST be prefixed with `ELASTIC_OTEL_`.
+For example, the [universal profiling integration](#universal-profiling-integration) can be enabled with `ELASTIC_OTEL_UNIVERSAL_PROFILING_INTEGRATION_ENABLED`.
+
+Elastic and platform specific configuration items must be prefixed with `ELASTIC_OTEL_${platform}_` to be consistent with
+the upstream `OTEL_${platform}_` prefix.
+
+For simplicity the configuration in this specification will use the "environment variable" syntax, some platforms like Java
+might also support other ways to configure.
+
+## Features
+
+### Inferred spans
+
+Supported platforms: [Java](https://github.com/elastic/elastic-otel-java/tree/main/inferred-spans)
+
+Configuration to enable: `ELASTIC_OTEL_INFERRED_SPANS_ENABLED`
+
+Note: While the implementation is Java-only for now, it should probably have been using `ELASTIC_OTEL_JAVA_INFERRED_SPANS_ENABLED`
+instead, but we plan to fix this inconsistency once it has been contributed upstream.
+
+### System metrics
+
+These metrics are usually captured using the collector running locally but in case where no collector is present, or a centralized
+collector is used then the user might opt in to also collect those.
+
+These metrics are not captured by default in order to prevent duplicated metrics when they are also captured by a collector.
+
+TODO : add link to supported platforms and respective implementations
+TODO : add the configuration option name to enable those
```

Comment:
Will you fill in these TODO`s before merging?

Reply:
I think this is still a TODO for every language :sweat_smile:

Reply:
The goal of this PR was to bootstrap the spec, so the goal was to merge them with the TODOs and add the missing pieces later.

Reply:
> What do you think about `ELASTIC_OTEL_SYSTEM_METRICS_ENABLED=false|true` ?

@xrmx that looks great to me, but let's do that in a separate follow-up PR for simplicity and avoid mixing too many things.

We haven't done that recently, but the usual process involved:
- creating a "master issue" with links to the individual tasks below
- one PR for updating the spec
- one issue per platform (this is automated)

However this hasn't been used in a while since the addition of otel agents, so you might have to do this manually.